pyccel/codegen/printing/ccode.py

```diff
@@ -791,8 +791,9 @@ def get_declare_type(self, expr):
                 if expr.rank > 15:
                     errors.report(UNSUPPORTED_ARRAY_RANK, severity='fatal')
                 self._additional_imports.add('ndarrays')
-                return 't_ndarray '
-            errors.report(PYCCEL_RESTRICTION_TODO, symbol="rank > 0",severity='fatal')
+                dtype = 't_ndarray'
```

Comment:
just a question, do we need here treating the case of declared array pointer `t_ndarray*`?

Reply:
Yes, this is important for optional arrays